modules/API/src/main/java/de/Keyle/MyPet/api/util/Timer.java

```diff
@@ -62,6 +65,36 @@ public static void startTimer() {
                 player.schedule();
             }
         }, 10L, 20L));
+
+        timerIDs.add(Bukkit.getScheduler().scheduleSyncRepeatingTask(MyPetApi.getPlugin(), () -> {
+            // If  DECREMENT_TIME_IF_PET_STORED is false then this task is useless. However if its later turned on and
+            // the configuration is reloaded then the task will allow the change to take effect without a server restart.
+            if(Configuration.Respawn.DECREMENT_TIME_IF_PET_STORED) {
+                for (MyPetPlayer player : MyPetApi.getPlayerManager().getMyPetPlayers()) {
+                    final MyPetPlayer owner = player;
+                    // FIXME: Is there a data structure that gets you all pets for an online player?
+                    //  reading it from disk every time may not be the best option
+                    MyPetApi.getRepository().getMyPets(owner, new RepositoryCallback<List<StoredMyPet>>() {
```

Comment:
This is not really an option especially for MySQL/MongoDB databases 🤔 Maybe we should store the online time of a player and only update the stored pets when they get loaded/displayed

Reply:
I see your point. So you want to move the same portion of code from a timer to a method that is called only when the stored pet is loaded/displayed. And we would decrement based on online time.

So what exactly would be those update points?.

Server shutdown, Player logging out, petswitch command, petadmin switch command.

Am i missing anything?